igvm/hypervisor_preferences.py

```diff
@@ -12,30 +12,43 @@
 # them reusable, even though most of them are not reused.  Some of the classes
 # are so simple that they could as well just be a function, but kept
 # as classes to have a consistent style.
-
+import math
+import sys
 from logging import getLogger
-
-from igvm.utils import LazyCompare
+from typing import Union, List, Optional
 
 log = getLogger(__name__)
 
 
-class InsufficientResource(object):
+class HypervisorPreference(object):
+    def get_preference(self, vm, hv) -> Union[float, bool]:
+        raise NotImplementedError('get_preference is not implemented')
+
+
+class InsufficientResource(HypervisorPreference):
     """Check a resource of hypervisor would be sufficient"""
-    def __init__(self, hv_attribute, vm_attribute, multiplier=1, reserved=0):
-        self.hv_attribute = hv_attribute
-        self.vm_attribute = vm_attribute
-        self.multiplier = multiplier  # TODO Use identical units in Serveradmin
-        self.reserved = reserved
 
-    def __repr__(self):
+    def __init__(
+        self,
+        hv_attribute: str,
+        vm_attribute: str,
+        multiplier: int = 1,
+        reserved: int = 0,
+    ) -> None:
+        self.hv_attribute: str = hv_attribute
+        self.vm_attribute: str = vm_attribute
+        # TODO Use identical units in Serveradmin
+        self.multiplier: int = multiplier
+        self.reserved: int = reserved
+
+    def __repr__(self) -> str:
         args = repr(self.hv_attribute)
         if self.reserved:
             args += ', reserved=' + repr(self.reserved)
 
         return '{}({})'.format(type(self).__name__, args)
 
-    def __call__(self, vm, hv):
+    def get_preference(self, vm, hv) -> Union[float, bool]:
         # Treat freshly created HVs always passing this check
```

Comment:
With False being evaluated to `0.0` it would not longer pass this check but fail no ?

Reply:
`False` is an alias for `0.0` which makes the preference fail, yes